raiden/tests/utils/smartcontracts.py

```diff
@@ -9,10 +10,20 @@
 from raiden.network.proxies.custom_token import CustomToken
 from raiden.network.proxies.service_registry import ServiceRegistry
 from raiden.network.rpc.client import JSONRPCClient
-from raiden.utils.typing import Any, Dict, List, TokenAmount, Tuple
+from raiden.utils.typing import Any, Dict, List, T_TransactionHash, TokenAmount, Tuple
 from raiden_contracts.contract_manager import ContractManager
 
 
+def is_tx_hash_bytes(bytes_: Any) -> bool:
+    """
+    Check wether the `bytes_` is a correctly encoded transaction hash,
+    but do not query any blockchain node to check for transaction validity.
+    """
+    if isinstance(bytes_, T_TransactionHash):
+        return bool(re.fullmatch("^0x([A-Fa-f0-9]{64})$", bytes_.hex()))
```

Comment:
Wouldn't it be easier to do
```suggestion
        return len(bytes_) == 32
```
?

Reply:
Ok, so first of all I had some misconception about bytes / hex encoding in python - you are completely right.

Second, it was not directly clear to me, that this is not a `bytes` instance, but rather a `hexbytes.HexBytes` instance, that is returned from web3.

Where the difference is in hex encoding:

```python
from hexbytes import HexBytes

b = b'6\x03\xe4Ne[lth\xd5\xd2#\xddw\xfd\x15\xa9i=\xd57\x8b\xf0Mf\xfe\xc1X\xf1\x7f\xb3\xeb'
hb = HexBytes(b)

print(hb.hex())
print(b.hex())
```
With output:
```
>>>>  '0x3603e44e655b6c7468d5d223dd77fd15a9693dd5378bf04d66fec158f17fb3eb'
>>>>  '3603e44e655b6c7468d5d223dd77fd15a9693dd5378bf04d66fec158f17fb3eb'
```
So I assumed this to be bytes, and `.hex()` encoding "accidentally" provided me with the 0x prefixed string in the API response:

https://github.com/ezdac/raiden/blob/5c15b68d8e6e6d3a67d3993aa8a4487ae137d996/raiden/api/rest.py#L588

So I guess the `is_tx_hash_bytes` not only tests the length of the underlying bytearray,
but also the return value of the `.hex()` method.

Would it make sense to change `T_TransactionHash` from `bytes` to `hexbytes.HexBytes`?

Reply:
This was resolved by implementing the suggestions and using `eth_utils.encode_hex( )` for encoding `TransactionHash` typed values.